pallets/parachain-staking/src/lib.rs

```diff
@@ -216,12 +422,37 @@ pub mod pallet {
 		}
 	}
 
-	impl<A: Clone, B: Copy> From<Collator<A, B>> for CollatorSnapshot<A, B> {
-		fn from(other: Collator<A, B>) -> CollatorSnapshot<A, B> {
+	impl<A: Clone + Ord, B: Ord + Copy> From<Collator<A, B>> for Collator2<A, B> {
+		fn from(other: Collator<A, B>) -> Collator2<A, B> {
```

Comment:
Nice! I like this pattern for migrating data types. Wish I had done that on the author mapping one.

Reply:
Needs a unit test to make sure it respects the ordering expected